### PR TITLE
A solved task has been added

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
+def long_time_func_2(n_tuple: tuple, power: int) -> list:
+    return [number ** power for number in n_tuple]
+
+
+print(long_time_func(1, 2, 3))
+print(long_time_func(2, 2, 3))
+print(long_time_func_2((5, 6, 7), 5))
+print(long_time_func(1, 2, 3))
+print(long_time_func_2((5, 6, 7), 10))
+print(long_time_func_2((5, 6, 7), 10))
```
Comment (viktoria-rybenchuk) [inline app/main.py]: This code is redundant. Make sure to remove all code for debugging purposes before pushing to GitHub

File: app/main.py [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_data = {}
+
+    @wraps(func)
+    def wrapper(*args) -> None:
```
Comment (viktoria-rybenchuk) [inline app/main.py]: use kwargs as well

File: app/main.py [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
+        if args in cache_data:
+            print("Getting from cache")
+            return cache_data[args]
+        else:
```
Comment (viktoria-rybenchuk) [inline app/main.py]: else is redundant

**Review by viktoria-rybenchuk — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 3 above)
Comment (blackbsdl) [reply app/main.py:13]: If I remove the else, then I can't figure out how to ensure that the conditions are met correctly?

**Review by pavlopro — APPROVED** [review]
GJ!

**Pull request closed without merging**